File: Go_leftToDo/tasks/checklistTask.go
package tasks

import (
	"fmt"
	"strings"
)

// Struct with embedded Task
type ChecklistTask struct {
	Task
}

// private Constructor function
func newChecklistTask(description string) (c ChecklistTask) {
	c.TaskType = "C"
	c.Done = false
	c.Description = description
	c.SubTask = []*SimpleTask{}
	return c
}

// Implement TaskRepository interface - adds input from user as description.
func (c ChecklistTask) Create() *ChecklistTask {
	var input string
	fmt.Printf("\nAnge Rubrik-uppgift:\n")
	fmt.Scanln(&input)

	checklist := newChecklistTask(strings.ToUpper(input))

	fmt.Println("Ange uppgifter, separerat med enter. [0] för att slutföra checklistan.")
	stopInput := false
	for !stopInput {
		var subInput string
		fmt.Scanln(&subInput)
		if subInput != "0" {
			sub := NewSimpleTask(subInput)
			checklist.SubTask = append(checklist.SubTask, &sub)
		} else if subInput == "0" {
			stopInput = true
		}
	}
	return &checklist
}

File: C#_LeftToDo/src/Tasks/ChecklistTask.cs

using System;
using System.Collections.Generic;

namespace LeftToDo.Tasks
{
    public class Checklist : Task // Derives from Task
    {
        // Constructor
        public Checklist(string specification)
        {
            type = "C";
            description = specification;
            subTask = new List<SimpleTask>();
        }

        // Adds input from user as description.
        internal override void Create()
        {
            Console.WriteLine("Ange Rubrik-uppgift:\n");
            description = Console.ReadLine().ToUpper();

            bool input = true;
            Console.WriteLine("Ange uppgifter, separerat med enter.[0] för att slutföra checklistan.");
            do
            {
                var specification = Console.ReadLine();
                if (specification == "0")
                {
                    input = false;
                }
                else
                {
                    var newTask = new SimpleTask(specification);
                    newTask.type = "sub";

                    subTask.Add(newTask);
                }
            } while (input);
        }
    }
}
